src/sentry/plugins/interfaces/releasehook.py

```diff
@@ -88,5 +89,21 @@ def finish_release(self, version, **values):
             datetime=values['date_released'],
         )
 
+        # check if user exists, and then try to get refs based on version
+        if values.get('owner', None):
+            try:
+                repo = Repository.objects.get(
+                    organization_id=self.project.organization_id)
+            except (Repository.MultipleObjectsReturned, Repository.DoesNotExist):
```

Comment:
let's not rely on the `MultipleObjectsReturned` exception. The problem here is that Django will return all rows from the database in this case anyways, which is too much for us.

So the pattern I liek to do in the case of checking for "0, 1 or more than 1", I'll do this:

```
repos = Repository.objects.filter(
    organization_id=self.project.organization_id,
)[0:2]
if len(repos) == 0:
    # none found
elif len(repos) == 1:
    # we found just one
else:
    # we found more than 1
```

This is minor, but it's lighter on the database. This just translates to doing a `LIMIT 2` against Postgres to make sure we don't return back everything when we don't need it.